### PR TITLE
Refactored to use JSON API on web

### PR DESCRIPTION
**Pull request by mattludwigs**

### CODE REVIEW

File: lib/vintage_net_wizard/backend/default.ex [hunk 1]
```diff
@@ -88,6 +88,7 @@ defmodule VintageNetWizard.Backend.Default do
         {VintageNet, ["interface", "wlan0", "wifi", "access_points"], _, access_points, _},
         state
       ) do
+    access_points = Enum.map(access_points, &Map.from_struct/1)
```
Comment (fhunleth) [inline lib/vintage_net_wizard/backend/default.ex]: Was the `__struct__` field causing issues?

File: priv/static/js/app.js [hunk 2]
```diff
@@ -24,143 +24,155 @@ const handle_scanned_ssid = function (data, table_id) {
...
-  var payload = JSON.parse(event.data);
-  switch (payload.type) {
-    case "wifi_scan": {
-      handle_scanned_ssid(payload.data, "wifi_scan");
-      break;
-    }
-    case "wifi_cfg": {
-      handle_scanned_ssid(payload.data, "wifi_cfg");
-      break;
+fetch("/api/v1/access_points")
```
Comment (fhunleth) [inline priv/static/js/app.js]: I'm not seeing where the `access_points` get polled for updates. Did I miss it or is it coming?

File: lib/vintage_net_wizard/backend/default.ex (hunk 1 above)
Comment (mattludwigs) [reply lib/vintage_net_wizard/backend/default.ex]: When calling `Jason.ecode` it expects the struct to have the `Jason.Encoder` protocol on it and I assumed that doing that would be more expected then implementing the protocol.

File: priv/static/js/app.js (hunk 2 above)
Comment (mattludwigs) [reply priv/static/js/app.js]: Forgot that piece I will do that.

File: lib/vintage_net_wizard/backend/default.ex (hunk 1 above)
Comment (fhunleth) [reply lib/vintage_net_wizard/backend/default.ex]: Makes sense.

**Pull request merged**